app/queue/egress.py

```diff
@@ -21,13 +22,12 @@ def __init__(self, config):
         self._topic = config.host_egress_topic
 
     def write_event(self, event, key, headers):
-        logger.debug("Topic: %s, key: %s, event: %s, headers: %s", self._topic, key, event, headers)
-
         try:
             k = key.encode("utf-8") if key else None
             v = event.encode("utf-8")
             h = [(hk, hv.encode("utf-8")) for hk, hv in headers.items()]
             self._kafka_producer.send(self._topic, key=k, value=v, headers=h)
+            log_produced_message(self._topic, event, key, headers)
```

Comment:
Since `send()` is asynchronous it would be better to register a callback here and only declare victory in the logs once we receive the ack.

```suggestion
            self._kafka_producer.send(self._topic, key=k, value=v, headers=h).add_callback(log_event_produced).add_errback(log_event_error)
```

Reply:
🤔